packages/ui/src/components/Heading/Heading.tsx

```diff
@@ -1,40 +1,89 @@
 import { HTMLAttributes, forwardRef } from "react";
+import { tv, type VariantProps } from "tailwind-variants";
 
-import { IconType, Color } from "../../types";
-import { headingStyles, iconStyles } from "./styles";
-import { HeadingLevel, HeadingRef } from "./types";
+import { IconType } from "../../types";
+import { HeadingRef } from "./types";
 
-export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
-  level?: HeadingLevel;
-  size?: HeadingLevel;
-  color?: Color;
-  Icon?: IconType;
+const heading = tv({
+  slots: {
+    base: "",
+    icon: "inline-block h-auto shrink-0 align-middle",
+  },
+  variants: {
+    hasIcon: {
+      true: "flex items-center gap-x-2.5",
+    },
+    size: {
+      h1: {
+        base: "text-5xl/[1.1] font-bold lg:text-6xl/[1.1]",
+        icon: "size-12 stroke-[1.5]",
+      },
+      h2: {
+        base: "mt-12 mb-3 text-4xl/[1.1] font-bold lg:text-5xl/[1.1]",
+        icon: "size-10 stroke-[1.6]",
+      },
+      h3: {
+        base: "mt-9 mb-1.5 text-3xl/[1.1] lg:text-4xl/[1.1]",
+        icon: "size-8 stroke-[1.6]",
+      },
+      h4: {
+        base: "mt-9 mb-1.5 text-2xl/[1.1] lg:text-3xl/[1.1]",
+        icon: "size-7 stroke-[1.6]",
+      },
+      h5: {
+        base: "mt-6 mb-1.5 text-xl/[1.1] lg:text-2xl/[1.1]",
+        icon: "size-6 stroke-[1.6]",
+      },
+      h6: {
+        base: "mt-6 mb-1.5 text-lg/[1.1] font-bold lg:text-xl/[1.1]",
+        icon: "size-5 stroke-[1.6]",
+      },
+    },
+    color: {
+      primary: {
+        icon: "text-primary",
+      },
+      secondary: {
+        icon: "text-secondary iap:dark:text-secondary-300",
+      },
+      success: {
+        icon: "text-success",
+      },
+      warning: {
+        icon: "text-warning",
+      },
+      error: {
+        icon: "text-error",
+      },
+    },
+  },
+});
+
+type HeadingVariants = VariantProps<typeof heading>;
+
+export interface HeadingProps
+  extends HeadingVariants,
```

Comment:
HeadingVariants includes `hasIcon` which we don't want to expose as a prop, right?  It doesn't get used on L80.